pkg/metrics/iterable_series.go

```diff
@@ -76,25 +73,31 @@ func (series *IterableSeries) WriteFooter(stream *jsoniter.Stream) error {
 
 // WriteCurrentItem writes the json representation of an item
 func (series *IterableSeries) WriteCurrentItem(stream *jsoniter.Stream) error {
-	if series.current == nil {
+	current := series.Current()
+	if current == nil {
 		return errors.New("nil serie")
 	}
-	return writeItem(stream, series.current)
+	return writeItem(stream, current)
 }
 
 // DescribeCurrentItem returns a text description for logs
 func (series *IterableSeries) DescribeCurrentItem() string {
-	if series.current == nil {
+	current := series.Current()
+	if current == nil {
 		return "nil serie"
 	}
-	return describeItem(series.current)
+	return describeItem(current)
 }
 
 // MoveNext advances to the next element.
 // Returns false for the end of the iteration.
 func (series *IterableSeries) MoveNext() bool {
-	var ok bool
-	series.current, ok = <-series.c
+	v, ok := series.c.Get()
+	if v != nil {
```

Comment:
`Get()` has the same semantic as a channel, when the boolean is false, the value returned is nil for a pointer.
Also note that when `MoveNext()` returns false, no value should be read so we can discard `v` anyway.

Reply:
Would it make sense to remove `ok` then, and make `Get` return a single value?

Reply:
@ogaca-dd I think I can see it from your point of view too so it's not a blocker for me but for me it is a bit of an odd thing from a black-box API standpoint to assume that there is a specific value of a return item when `ok` is false. As for discarding `v` on `ok == false` in the caller, we would be relying on an additional layer of logic surrounding this function to ensure proper operation, making the code less-than-ideal in modularity. Either way, this isn't a big enough issue to block the review so I think this is resolved from my point of view but I'll leave it as-is since @vickenty also left a note on this convo.

Reply:
I think this is the "go way" to handle this situation, for example checking if the channel is closed or check if something exist in a map (There are 2 values and the second one is a `bool`). I agree that it would be better to have something like a `Option<T>` but there is no such thing in Go. Also I would like to be as closed as possible as the channel interface and not having `IsClosed` function for example.